lighthouse-core/computed/resource-summary.js

```diff
@@ -51,7 +51,7 @@ class ResourceSummary {
       'third-party': {count: 0, size: 0},
     };
     const budget = Budget.getMatchingBudget(context.settings.budgets, mainResourceURL);
-    let firstPartyHosts = /** @type {Array<string>} */ ([]);
+    let firstPartyHosts = /** @type {Immutable<Array<string>>} */ ([]);
```

Comment:
this will be able to be just `ReadonlyArray<string>` after we upgrade to tsc 3.7+